stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -2575,6 +2575,7 @@ function setindex!(A::SparseMatrixCSC{Tv,Ti}, V::AbstractVecOrMat, Ix::Union{Int
     @assert !has_offset_axes(A, V, Ix, Jx)
     (I, J) = Base.ensure_indexable(to_indices(A, (Ix, Jx)))
     checkbounds(A, I, J)
+    Base._setindex!(IndexStyle(A), A, V, to_indices(A, (Ix, Jx))...)
```

Comment:
Doesn't this just assign into the array twice?  Once with the generic (slow) dense version, and then once with the optimized sparse version?  We just want to be checking the shape of `V`, no?

Reply:
Indeed: before:

```
julia> A = spzeros(1000,1000)
1000×1000 SparseMatrixCSC{Float64,Int64} with 0 stored entries

julia> @btime $A[:,:] = $(spzeros(1000, 1000))
  5.557 μs (5 allocations: 8.39 KiB)
```

After:

```
julia> @btime $A[:,:] = $(spzeros(1000, 1000))
  9.545 ms (5 allocations: 8.39 KiB)
```

Reply:
#30923